tests/navier_stokes/poiseuille_flow_2d.cpp

```diff
@@ -0,0 +1,303 @@
+        time_integrator->initializePatchHierarchy(patch_hierarchy, gridding_algorithm);
+
+        // Deallocate initialization objects.
+        app_initializer.setNull();
+
+        // Print the input database contents to the log file.
+        plog << "Input database:\n";
+        input_db->printClassData(plog);
+
+        // Write out initial visualization data.
+        int iteration_num = time_integrator->getIntegratorStep();
+        double loop_time = time_integrator->getIntegratorTime();
+        if (dump_viz_data && uses_visit)
+        {
+            pout << "\n\nWriting visualization files...\n\n";
+            time_integrator->setupPlotData();
+            visit_data_writer->writePlotData(patch_hierarchy, iteration_num, loop_time);
+        }
+
+        // Open streams to save position and velocity of the structure, as well
+        // as the volume of the two phase and the Lagrange multiplier.
+        ofstream vol_stream;
+        if (SAMRAI_MPI::getRank() == 0)
+        {
+            vol_stream.open("output", ios_base::out | ios_base::app);
+            vol_stream.precision(16);
+            vol_stream.setf(ios::fixed, ios::floatfield);
+        }
+
+        // Main time step loop.
+        double loop_time_end = time_integrator->getEndTime();
+        double dt = 0.0;
+        while (!IBTK::rel_equal_eps(loop_time, loop_time_end) && time_integrator->stepsRemaining())
+        {
+            iteration_num = time_integrator->getIntegratorStep();
+            loop_time = time_integrator->getIntegratorTime();
+
+            pout << "\n";
+            pout << "+++++++++++++++++++++++++++++++++++++++++++++++++++\n";
+            pout << "At beginning of timestep # " << iteration_num << "\n";
+            pout << "Simulation time is " << loop_time << "\n";
+
+            dt = time_integrator->getMaximumTimeStepSize();
+            time_integrator->advanceHierarchy(dt);
+            loop_time += dt;
+
+            pout << "\n";
+            pout << "At end       of timestep # " << iteration_num << "\n";
+            pout << "Simulation time is " << loop_time << "\n";
+            pout << "+++++++++++++++++++++++++++++++++++++++++++++++++++\n";
+            pout << "\n";
+
+            // Update the target volume due to the inflow of liquid through the specified boundary (= y_bottom)
+            VariableDatabase<NDIM>* var_db = VariableDatabase<NDIM>::getDatabase();
+            const int u_idx = var_db->mapVariableAndContextToIndex(time_integrator->getVelocityVariable(),
+                                                                   time_integrator->getCurrentContext());
+
+            const double vol_inflow = dt * LevelSetUtilities::computeNetInflowPhysicalBoundary(
+                                               time_integrator->getHierarchyMathOps(),
+                                               u_idx,
+                                               /*location_idx*/ 0); // location_idx is boundary_0
+
```

Comment:
Do we need to multiply by `dt` here? I think we can just compute `vol_flow_rate` here. The advantage is that it will be equal to `U_max` that is set in the input file; easy to check for the accuracy.

Reply:
You would need to update the output file also.